KakapoExample/KakapoTests/JSONAPITests.swift

```diff
@@ -0,0 +1,270 @@
+        
+        describe("JSON API Entity relationship serialization") {
+            it("should serialize the relationships when they are single JSONAPIEntities") {
+                let object = json(user)
+                let dog = object["relationships"]["dog"]["data"]
+                expect(dog.dictionary).toNot(beNil())
+                expect(dog["id"].stringValue).to(equal("22"))
+                expect(dog["type"].stringValue).to(equal("dog"))
+            }
+            
+            it("should serialize the relationships when they are arrays of JSONAPIEntities") {
+                let object = json(user)
+                let cats = object["relationships"]["cats"]["data"].array!
+                expect(cats.count).to(equal(2))
+                expect(cats[0]["id"].stringValue).to(equal("33"))
+                expect(cats[0]["type"].stringValue).to(equal("cat"))
+                expect(cats[1]["id"].stringValue).to(equal("44"))
+                expect(cats[1]["type"].stringValue).to(equal("cat"))
+            }
+            
+            it("should not serialize relationships of relationships") {
+                let object = json(user)
+                let dogData = object["relationships"]["dog"]["data"].dictionary
+                expect(dogData).toNot(beNil())
+                expect(dogData?["relationships"]).to(beNil())
+            }
+            
+            it("should not serialize attributes of relationships") {
+                let object = json(user)
+                let dogData = object["relationships"]["dog"]["data"].dictionary
+                expect(dogData).toNot(beNil())
+                expect(dogData?["attributes"]).to(beNil())
+            }
+            
+            it("should not serialize nil relationships") {
+                let object = json(Dog(id: "22", name: "Joan", cat: nil))
+                let relationships = object["relationships"].dictionaryValue
+                expect(relationships["cat"]).to(beNil())
+                let attributes = object["attributes"].dictionaryValue
+                expect(attributes["cat"]).to(beNil())
+            }
+            
+            it("should serialize the relationships even when an array is empty") {
+                let lonelyMax = User(id: "11", name: "Max", dog: dog, cats: [])
+                let object = json(lonelyMax)
+                let cats = object["relationships"]["cats"].dictionary!
+                expect(cats.count).to(equal(1))
+                let dataArray = cats["data"]
+                expect(dataArray).toNot(beNil())
+                expect(dataArray?.count).to(equal(0))
+            }
+            
+            context("Optional relationships") {
+                it("should not be included when nil") {
+                    let object = json(Dog(id: "123", name: "A", cat: nil))
+                    let relationships = object["relationships"].dictionaryObject
+                    expect(relationships).to(beNil())
+                }
```

Comment:
This test is mostly the same as `should not serialize nil relationships` I think, but I'm fine with leaving both